plugins/inputs/chrony/chrony.go

```diff
@@ -89,13 +133,17 @@ func processChronycOutput(out string) (map[string]interface{}, map[string]string
 	fields := map[string]interface{}{}
 	lines := strings.Split(strings.TrimSpace(out), "\n")
 	for _, line := range lines {
+		if len(strings.TrimSpace(line)) == 0 {
+			// When running multiple commands, there are sometimes empty lines between output
+			continue
+		}
 		stats := strings.Split(line, ":")
 		if len(stats) < 2 {
 			return nil, nil, fmt.Errorf("unexpected output from chronyc, expected ':' in %s", out)
 		}
 		name := strings.ToLower(strings.Replace(strings.TrimSpace(stats[0]), " ", "_", -1))
-		// ignore reference time
-		if strings.Contains(name, "ref_time") {
+		// ignore reference time - different keys for "tracking" vs "ntpdata" commands
+		if strings.Contains(name, "ref_time") || strings.Contains(name, "reference_time") {
```

Comment:
I feel like the method we are using to parse the output of chrony is too general, and may cause trouble now that there are other commands we are parsing.  It would be best to have a separate parser for each command output, so that we can have special processing for each command and so that we can set the names and types by hand.

The first thing I would put in place is some test cases for each command you would like to support.